src/Service/CollapsedBroadcastsService.php

```diff
@@ -102,145 +137,198 @@ public function findByCategoryAndStartAtDateRange(
+                return $result;
             }
-
-            $result[$year][] = (int) $period['month'];
-        }
-
-        return $result;
+        );
     }
 
     public function findDaysByCategoryInDateRange(
         Category $category,
         DateTimeImmutable $start,
-        DateTimeImmutable $end
+        DateTimeImmutable $end,
+        $ttl = CacheInterface::NORMAL
     ): array {
-        $rows = $this->repository->findBroadcastedDatesForCategory(
-            $category->getDbAncestryIds(),
-            false,
-            $start,
-            $end
-        );
-
-        $result = [];
+        $key = $this->cache->keyHelper(__CLASS__, __FUNCTION__, $category->getDbId(), $start->getTimestamp(), $end->getTimestamp(), $ttl);
+
+        return $this->cache->getOrSet(
+            $key,
+            $ttl,
+            function () use ($category, $start, $end) {
+                $rows = $this->repository->findBroadcastedDatesForCategory(
+                    $category->getDbAncestryIds(),
+                    false,
+                    $start,
+                    $end
+                );
+
+                $result = [];
+
+                foreach ($rows as $row) {
+                    $year = (int) $row['year'];
+                    if (!isset($result[$year])) {
+                        $result[$year] = [];
+                    }
 
-        foreach ($rows as $row) {
-            $year = (int) $row['year'];
-            if (!isset($result[$year])) {
-                $result[$year] = [];
-            }
+                    $month = (int) $row['month'];
+                    if (!isset($result[$year][$month])) {
+                        $result[$year][$month] = [];
+                    }
 
-            $month = (int) $row['month'];
-            if (!isset($result[$year][$month])) {
-                $result[$year][$month] = [];
-            }
+                    $day = (int) $row['day'];
+                    if (!in_array($day, $result[$year][$month])) {
+                        $result[$year][$month][] = $day;
+                    }
+                }
 
-            $day = (int) $row['day'];
-            if (!in_array($day, $result[$year][$month])) {
-                $result[$year][$month][] = $day;
+                return $result;
             }
-        }
-
-        return $result;
+        );
     }
 
     public function filterCategoriesByBroadcastedDate(
         array $allCategories,
         DateTimeImmutable $from,
-        DateTimeImmutable $to
+        DateTimeImmutable $to,
+        $ttl = CacheInterface::NORMAL
     ): array {
         if (empty($allCategories)) {
             return [];
         }
 
-        $categoriesAncestryIds = [];
-        foreach ($allCategories as $category) {
-            $categoriesAncestryIds[] = $category->getDbAncestryIds();
-        }
+        $key = $this->cache->keyHelper(__CLASS__, __FUNCTION__, md5(serialize($allCategories)), $from->getTimestamp(), $to->getTimestamp(), $ttl);
```

Comment:
Build an array of the category Ids to use in the key, rather than `md5(serialize($allCategories))`.

In the segments service you do this imploding on '-'.